src/Http/Client/Response.php

```diff
@@ -442,7 +444,41 @@ public function getCookieData($name)
             return null;
         }
 
-        return $this->cookies->get($name)->toArrayClient();
+        $cookie = $this->cookies->get($name);
+
+        return $this->toArrayClient($cookie);
+    }
+
+    /**
+     * Convert the cookie into an array of its properties.
+     *
+     * This method is compatible with older client code that
+     * expects date strings instead of timestamps.
+     *
+     * @param \Cake\Http\Cookie\CookieInterface $cookie Cookie object.
+     * @return array
+     */
+    protected function toArrayClient(CookieInterface $cookie)
+    {
+        if ($cookie instanceof Cookie) {
+            return $cookie->toArrayClient();
+        }
+
+        if ($cookie->getExpiry()) {
+            $expires = $cookie->getExpiry()->format(Cookie::EXPIRES_FORMAT);
```

Comment:
@markstory I'm not sure. The cookie returns `DateTimeInterface` so you can format it anyway you like.